src/update/update.go

```diff
@@ -68,10 +69,10 @@ func CheckAndUpdate(config *core.Configuration, updatesEnabled, updateCommand, f
 		fmt.Fprintf(os.Stderr, "%s Please from version %s to %s", word, pleaseVersion(), config.Please.Version.VersionString())
 	}
 
-	// Must lock here so that the update process doesn't race when running two instances
-	// simultaneously.
-	core.AcquireRepoLock()
-	defer core.ReleaseRepoLock()
+	// Must lock exclusively here so that the update process doesn't race when running two instances simultaneously.
+	// Once we are done we replace/restore the mode to the shared one.
+	core.AcquireExclusiveRepoLock()
+	defer core.AcquireSharedRepoLock()
```

Comment:
Should this be `core.ReleaseExclusiveRepoLock()` instead ?

Reply:
No, as it would release the lock making it prone to race conditions with other processes. We want to keep the lock open for the lifetime of the process and just swap between the modes (ie. exclusive and shared) as needed. The shared mode should always be the default one. You can almost think of it as the lock being "released" in one sense, but it brings the benefits described in an earlier comment.